rx-preferences/src/main/java/com/f2prateek/rx/preferences2/RxSharedPreferences.java

```diff
@@ -167,9 +167,11 @@ public void onSharedPreferenceChanged(SharedPreferences preferences, String key)
       @NonNull Set<String> defaultValue) {
     checkNotNull(key, "key == null");
     checkNotNull(defaultValue, "defaultValue == null");
-    return new RealPreference<>(preferences, key, defaultValue, StringSetAdapter.INSTANCE, keyChanges);
+    return new RealPreference<>(preferences, key,
+        Collections.unmodifiableSet(new LinkedHashSet<>(defaultValue)), StringSetAdapter.INSTANCE,
```

Comment:
good point!
I agree consistently returning the same reference for all default values outweighs consistently returning an unmodifiable string set.